File: bindings/go/examples/get_object.go
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

package main

import (
	"encoding/hex"
	"fmt"
	"log"

	sdk "bindings/iota_sdk_ffi"
)

func isNilError(err error) bool {
	if sdkErr, ok := err.(*sdk.SdkFfiError); ok {
		return sdkErr == nil
	}
	return false
}

func main() {
	client := sdk.GraphQlClientNewDevnet()

	objectID, err := sdk.ObjectIdFromHex("0x20c056090c3dd1604fcfd7ea759781de650aa45323738e799365d0c28bebeb1e")
	if err != nil {
		log.Fatalf("Failed to parse object ID: %v", err)
	}

	objOpt, err := client.Object(objectID, nil)
	if !isNilError(err) {
		log.Fatalf("Failed to get object contents: %v", err)
	}
	if objOpt == nil {
		log.Fatal("Missing object")
	}
	obj := *objOpt

	objType := "Package"
	if obj.ObjectType().IsStruct() { 
		objType = obj.ObjectType().AsStruct().String()
	}

	objOwner := "Immutable"
	if obj.Owner().IsAddress() {
		objOwner = fmt.Sprintf("Address(%v)", obj.Owner().AsAddress().ToHex())
	} else if obj.Owner().IsObject() {
		objOwner = fmt.Sprintf("Object(%v)", obj.Owner().AsObject().ToHex())
	} else if obj.Owner().IsShared() {
		objOwner = fmt.Sprintf("Shared(%v)", obj.Owner().AsShared())
	}

	fmt.Println("Object ID:", obj.ObjectId().ToHex())
    fmt.Println("Version:", obj.Version())
    fmt.Println("Previous transaction:", obj.PreviousTransaction().ToBase58())
    fmt.Println("Owner:", objOwner)
    fmt.Println("Storage rebate:", obj.StorageRebate())
    fmt.Println("Type:", objType)
    fmt.Println("BCS bytes:", hex.EncodeToString(obj.AsStruct().Contents))

}

File: bindings/kotlin/examples/GetObject.kt
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import iota_sdk.GraphQlClient
import iota_sdk.ObjectId
import iota_sdk.hexEncode
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    try {

        val client = GraphQlClient.newTestnet()

        val objectId =
            ObjectId.fromHex("0x541b117cac18fb1c07a293db300acd12b05c01fa81232b37151b005ca7d4f755")

        val obj = client.`object`(objectId)!!

        println("Object ID: ${obj.id().toHex()}")
        println("Version: ${obj.version()}")
        println("Previous transaction: ${obj.previousTransaction().toBase58()}")
        println("Owner: ${obj.owner().toString()}")
        println("Storage rebate: ${obj.storageRebate()}")
        println("Type: ${obj.objectType().toString()}")
        @OptIn(kotlin.ExperimentalStdlibApi::class)
        println("BCS bytes: ${hexEncode(obj.asStruct().contents)}")
    } catch (e: Exception) {
        e.printStackTrace()
        kotlin.system.exitProcess(1)
    }
}
